### PR TITLE
ISPN-8038 java.lang.ClassCastException with compatibility mode and te…

### PR DESCRIPTION
**Pull request by gustavonalle**
…xt content

https://issues.jboss.org/browse/ISPN-8038

### CODE REVIEW

File: commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.infinispan.commons.marshall;
+
+import java.nio.charset.Charset;
+
+public final class UTF8Marshaller extends StringMarshaller {
+
+   public UTF8Marshaller() {
+      super(Charset.forName("UTF-8"));
```
Comment (tristantarrant) [inline commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java]: Use java.nio.charset.StandardCharsets.UTF_8 instead (since Java 1.7)

**Comment by tristantarrant** [comment]
Is this PR causing http://ci.infinispan.org/job/Infinispan/job/PR-5278/3/testReport/junit/org.infinispan.it.compatibility/DistMemcachedEmbeddedTest/setup/ ?

**Comment by gustavonalle** [comment]
@tristantarrant I don't think so  http://ci.infinispan.org/job/Infinispan/job/master/23/testReport/junit/org.infinispan.it.compatibility/ByteArrayValueReplEmbeddedHotRodTest/setup/

File: commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.infinispan.commons.marshall;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class UTF8Marshaller extends StringMarshaller {
```
Comment (anistor) [inline commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java]: There are 2 usages of ```new StringMarshaller(Charset.forName("UTF-8"))``` that should be replaced by this class.

Comment (anistor) [inline commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java]: Maybe UTF8StringMarshaller would be a more suitable name.

File: commons/src/main/java/org/infinispan/commons/dataconversion/CompatModeEncoder.java [hunk 3]
```diff
@@ -33,7 +33,7 @@ public Object toStorage(Object content) {
    public Object fromStorage(Object content) {
```
Comment (anistor) [inline commons/src/main/java/org/infinispan/commons/dataconversion/CompatModeEncoder.java]: The toStorage/fromStorage methods should be annotated with ```@Override```. This is missing in some other implementations of ```Encoder```.

File: commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java (hunk 2 above)
Comment (gustavonalle) [inline commons/src/main/java/org/infinispan/commons/marshall/UTF8Marshaller.java]: +1

**Comment by gustavonalle** [comment]
All suggestions applied.

**Comment by anistor** [comment]
Thanks, looks good. After the latest changes some unused imports were left in ExecTypedTest and CustomMemcachedHotRodTest. I'll fix them and merge.

**Comment by anistor** [comment]
Merged in master. Thanks @gustavonalle !

**Pull request closed without merging**